src/main/java/org/folio/mappers/FolioToRtacMapper.java

```diff
@@ -25,23 +39,106 @@ public RtacHoldings mapToRtac(InventoryHoldingsAndItems instance) {
   }
 
+  private final Function<Item, RtacHolding> fromItemToRtacHolding =
+      item ->
+          new RtacHolding()
+              .withId(item.getId())
+              .withLocation(mapLocation(item))
+              .withCallNumber(mapCallNumber(item))
+              .withStatus(item.getStatus())
+              .withTemporaryLoanType(item.getTemporaryLoanType())
+              .withPermanentLoanType(item.getPermanentLoanType())
+              .withDueDate(item.getDueDate())
+              .withVolume(mapVolume(item));
+
+  /**
+   * This function is populating holding-level information for periodicals.
+   * The following php script was taken as an example:
+   * function holdingsLoad($response){
+   * $data = json_decode($response,true);
+   * $holdings = $data['holdingsRecords'];
+   * echo '<holdings>';
+   * foreach ($holdings as $holding){
+   * $statement = $holding['holdingsStatements'][0]['statement'];
+   * if ($statement == null){
+   * $statement = 'Multi';
+   * };
+   * echo '
+   * <holding>
+   * <id>'.$holding['id'] .'</id>
+   * <callNumber>'.htmlspecialchars($holding['callNumber']).'</callNumber>
+   * <location>'.discoveryName($holding['permanentLocationId']).'</location>
+   * <status>'.$statement.'</status>
+   * <dueDate></dueDate>
+   * </holding>';
+   * };
+   * echo '</holdings>';
+   * }
+   *
+   * @param holding - folio holding
+   * @return holding information to be returned to the caller
+   * @see <a href=https://issues.folio.org/browse/MODRTAC-37>MODRTAC-37</a>
+   */
+  private final Function<Holding, RtacHolding> fromHoldingToRtacHolding =
+      holding ->
+          new RtacHolding()
+              .withId(holding.getId())
+              .withCallNumber(mapCallNumber(holding))
+              .withLocation(mapLocation(holding))
+              .withStatus(mapHoldingStatements(holding))
+              .withDueDate(null);
+
+  private String mapHoldingStatements(Holding holding) {
+    final var holdingsStatements = holding.getHoldingsStatements();
+    var result = "Multi";
+    if (!holdingsStatements.isEmpty()) {
+      result = holdingsStatements.get(0).getStatement();
+    }
+    return result;
+  }
+
+  private String mapCallNumber(Holding holding) {
+    return holding.getCallNumber().getCallNumber();
+  }
+
+  private String mapCallNumber(Item item) {
+    final var callNumber = item.getCallNumber();
+    return assembleCallNumber(
+      callNumber.getCallNumber(), callNumber.getPrefix(), callNumber.getSuffix());
+  }
+
+  private String mapLocation(Holding holding) {
+    return holding.getLocation().getPermanentLocation().getName();
+  }
+
+  private String mapLocation(Item item) {
+    return item.getLocation().getLocation().getName();
+  }
+
+  private boolean isPeriodical(InventoryHoldingsAndItems instance) {
+    return isPeriodicalByNatureOfContent(instance)
+        || Objects.equals(instance.getModeOfIssuance(), "serial");
```

Comment:
again, isn't mode-of-issuance a controlled vocabulary?  Hardcoding "serial" here makes this fragile?

Reply:
You can't assume the reference data will always be present.  Some (actually most) institutions have chosen to not use the OOTB FOLIO reference data

Reply:
As it was stated in the requirements in MODRTAC-37, an instance is considered to be a  periodical when either:
* mode of issuance"= serial
* or  "nature of content" is one of (journal, newspaper).

I think, even if theese values are absent in a tenant, there will be no error, the system just would not be able to tell if an instance is a periodical or not.

Can you suggest a solution here?